File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        
        l = 0
        r = len(nums) - 1

#         if target <= nums[l]:
#             return l

#         if target > nums[r]:
#             return r+1

        while(l<=r):

            midpoint = (l+r)//2

            if nums[midpoint] == target:
                return midpoint

            elif nums[midpoint] < target:
                l = midpoint + 1

            else:
                r = midpoint - 1

        return l

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
		
		int start_point = 0;
		int end_point = nums.length - 1;
		
		while(start_point<=end_point) {
			
			int mid_length = (start_point+end_point)/2;
			
			if(target <= nums[mid_length]) {
				end_point = mid_length - 1;
			}
			else if(target > nums[mid_length]){
				start_point += 1;
			} else {
				return mid_length;
			}
		}
		
		return start_point;
	}
}